### ISSUE TITLE
Trainer Bug: update_every Tensorboard

### ISSUE THREAD

**Issue by haroldrubio** [haroldrubio/PytorchTrainer#19]
`update_every` unconditionally saves validation loss. Make it save the testing/validation loss depending on training mode.
